tests/unit/components/UrlBar.spec.js

```diff
@@ -21,6 +21,72 @@ describe("Given a UrlBar", () => {
       expect(urlBar.find("input").element.placeholder).to.equal(placeholder);
     });
   });
+
+  describe("When a valid URL is typed into the UrlBar", () => {
+    let input;
+
+    beforeEach(() => {
+      input = urlBar.find("input");
+
+      input.element.value = "http://www.test.com";
+    });
+
+    describe("And the <Enter> key is pressed in the UrlBar", () => {
+      beforeEach(() => {
+        input.trigger("keyup.enter");
+      });
+
+      it("Then the input element should be registered as valid", () => {
+        const isInputValid = !input.element.classList.contains("invalid");
+
+        expect(isInputValid).to.be.true;
+      });
+    });
+  });
+
+  describe("When an invalid URL is typed into the UrlBar", () => {
+    let input;
+
+    beforeEach(() => {
+      input = urlBar.find("input");
+
+      input.element.value = "abc";
+    });
+
+    describe("And the <Enter> key is pressed in the UrlBar", () => {
+      beforeEach(() => {
+        input.trigger("keyup.enter");
+      });
+
+      it("Then the input element should be registered as invalid", () => {
+        const isInputValid = !input.element.classList.contains("invalid");
+
+        expect(isInputValid).to.be.false;
+      });
+
+      it("Then no request is made", () => {
+        const emittedEvents = urlBar.emitted();
+
+        const wasEventEmittedFromInput = Object.prototype.hasOwnProperty.call(
```

Comment:
Is there a reason you don't just use `emittedEvents.hasOwnProperty("input")`?

Reply:
I actually tried that first, but it didn't work (I believe it returned `undefined`). When I tried `typeof urlBar.emitted()`, I got `object`. So I'm not quite sure what's going on there.

When I tried the other way, it worked, so I went with it (the `Object.prototype...` method)